### PR TITLE
Mark old cases

### PR DESCRIPTION
**Pull request by gamma032steam**
Cases older than 14 days (or where the data is inconsistent and we can't tell) are marked in blue. 

firefox_BcB8vMrAVE

### CODE REVIEW

**Comment by freddiewanah** [comment]
looks good to me. @morpheusp2p ?

File: src/ConfirmedMap.js [hunk 1]
```diff
@@ -20,11 +21,38 @@ class MbMap extends React.Component {
         };
     }
 
+    // Check if a date was more than two weeks ago
+    isOld(date) {
+        // Assume the 'Day DD/MM/YYYY' format
+        const dateComponents = date.split(" ");
+        const eventDay = dateComponents[1].split("/");
```
Comment (freddiewanah) [inline src/ConfirmedMap.js]: I'm wondering if there's no space in date, Probably would throw err here

Comment (gamma032steam) [reply src/ConfirmedMap.js]: Good catch, it does. I've fixed it and moved it all inside the try block. Thankfully no dates without spaces in the data. The date format is inconsistent so I'll talk with data about a convention.

Comment (freddiewanah) [reply src/ConfirmedMap.js]: Yeah, I've also talk to Ben about it. If the data side can fix the format issue, let's merge then

Comment (gamma032steam) [reply src/ConfirmedMap.js]: My original commit has a fix to the format of data points in the past two weeks, so they should all display accurately.

Comment (freddiewanah) [reply src/ConfirmedMap.js]: Yeah, I saw your commit, really appreciate that you've changed them by yourself. But the data team would update data every day [here](https://github.com/covid-19-au/covid-19-au.github.io/pull/73). So for future concern, we need to make sure their data match the frontend data format.

Comment (gamma032steam) [reply src/ConfirmedMap.js]: Spoke to Ben about this today. Will update tonight to handle a different, consistent data format for date.

**Comment by gamma032steam** [comment]
@freddiewanah Discussed with Ben that we are switching to DD/MM/YY format in data/mapdataCon.json. Ready for review when that gets updated.

File: src/ConfirmedMap.js [hunk 2]
```diff
@@ -20,11 +21,41 @@ class MbMap extends React.Component {
         };
     }
 
+    // Check if a date was more than two weeks ago
+    isOld(date) {
+        // Working with raw data, so try-catch just in case
+        try {
```
Comment (freddiewanah) [inline src/ConfirmedMap.js]: haha this is also what I thought, just push the whole process into a try function

File: src/ConfirmedMap.js [hunk 3]
```diff
@@ -20,11 +21,41 @@ class MbMap extends React.Component {
...
+            // 'DD/MM/YY' format
+            // Assume entries with incorrect formats are old
+            const eventDay = date.split("/");
+            if (eventDay.length !== 3 || eventDay === 'N/A') { return true; }
+
+            // Default constructor has current time
+            const today = new Date();
+            
+            // Day of the event. Transform to YYYY/MM/DD format
+            const day = eventDay[0].length == 1 ? '0' + eventDay[0] : eventDay[0];
```
Comment (freddiewanah) [inline src/ConfirmedMap.js]: why do we need the 0? I think `new Date("2020-3-23")` also works

File: src/data/mapdataCon.json [hunk 4]
```diff
@@ -14,7 +15,7 @@
     "state": "NSW",
     "area": "Sydney - Ryde Civic Centre",
     "name": "Sydney Church of Christ - Ryde Civic Centre: 7 cases",
-    "date": "8/3/20",
+    "date": "Saturday 21/03/2020",
```
Comment (freddiewanah) [inline src/data/mapdataCon.json]: I think you don't need the update here anymore

File: src/ConfirmedMap.js (hunk 3 above)
Comment (gamma032steam) [reply src/ConfirmedMap.js]: TIL

**Pull request merged**